extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java

```diff
@@ -521,9 +524,42 @@ public ValidationPhaseBuildItem validate(ObserverRegistrationPhaseBuildItem obse
                 unremovableBeans.stream().map(UnremovableBeanBuildItem::getPredicate).collect(Collectors.toList()));
         BeanDeploymentValidator.ValidationContext validationContext = beanProcessor.validate(bytecodeTransformerConsumer);
 
+        validatePostConstructAndPreDestroyMethods(index.getIndex());
+
         return new ValidationPhaseBuildItem(validationContext, beanProcessor);
     }
 
+    /**
+     * Checks that methods annotated with {@code PostConstruct} and {@code PreDestroy} are not returning {@code Uni}
+     * or {@code Multi}
+     */
+    void validatePostConstructAndPreDestroyMethods(IndexView index) {
```

Comment:
I wonder if we should throw the definition exception in `io.quarkus.arc.processor.Beans.getCallbacks()` and `io.quarkus.arc.processor.Beans.collectCallbacks()` instead... A callback method must have the following signature: `void <METHOD>()`. The current behavior is that we ignore `@PostConstruc/@PreDestroy` methods with a different signature.

Reply:
Wouldn't it be a breaking change for method returning something (even if ignored)?

Reply:
This PR is a partial breaking change, as the method is called, but everything happening "in" the Uni would not happen. 

We can make it stricter and say that method must return "void".

Reply:
> This PR is a partial breaking change, as the method is called, but everything happening "in" the Uni would not happen.

I don't understand - this pr throws a `DefinitionException` and the build fails, or?

`@PostConstruct Uni<Foo> init()` is currently ignored.

Reply:
I'm wondering about:

```
            if (method.returnType().kind() == Kind.VOID && method.parameterTypes().isEmpty()) {
```

in `Beans.collectCallbacks()`. 

Can't these methods receive injected beans?

Reply:
No. A lifecycle callback method defined on the bean class must not accept any params.

Reply:
Ah ok, so yes, we can move the check here.

Reply:
Check moved, improved error message.